src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1521,7 +1522,16 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, byte[]? p
 
                         gasAvailable -= GasCostOf.Base;
 
-                        result = txCtx.Header.BaseFeePerGas;
+                        result = blkCtx.Header.BaseFeePerGas;
+                        stack.PushUInt256(in result);
+                        break;
+                    }
+                case Instruction.BLOBBASEFEE:
+                    {
+                        if (!spec.BlobBaseFeeEnabled) goto InvalidInstruction;
+
+                        gasAvailable -= GasCostOf.Base;
+                        result = blkCtx.BlobBaseFee ?? throw new Exception("BlobBaseFee is not set. EIP-4844 has to be enabled for this opcode");
```

Comment:
don't throw base exception, go to Invalid Instruction?

Reply:
i am throwing because it is a problem with chainspec, where 7516 is enabled but 4844 is not. I want a message that will be displayed for people with incorrect chainspec file

Reply:
Maybe disallow enabling 7516 without 4844 in chainspec itself? We already have this kind of rules.
Also if you really want to throw then throw something meaningful like `InvalidOperationException`, never base `Exception`.

Reply:
done